_search-plugins/retrieve-specific-fields.md

```diff
@@ -0,0 +1,877 @@
+        }
+      ]
+    }
+    ```
+
+3. Perform a search with `inner_hits` and `docvalue_fields`:
+    ```
+    POST my_index/_search
+    {
+      "query": {
+        "nested": {
+          "path": "comments",
+          "query": {
+            "match_all": {}
+          },
+          "inner_hits": {
+            "docvalue_fields": ["username", "created_at"]
+          }
+        }
+      }
+    }
+    ```
+
+Expected response:
+
+```
+{
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1.0,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 1.0,
+        "_source": {
+          "title": "OpenSearch Basics",
+          "author": "John Doe",
+          "comments": [
+            {
+              "username": "alice",
+              "content": "Great article!",
+              "created_at": "2023-01-01T12:00:00Z"
+            },
+            {
+              "username": "bob",
+              "content": "Very informative.",
+              "created_at": "2023-01-02T12:00:00Z"
+            }
+          ]
+        },
+        "inner_hits": {
+          "comments": {
+            "hits": {
+              "total": {
+                "value": 2,
+                "relation": "eq"
+              },
+              "max_score": 1.0,
+              "hits": [
+                {
+                  "_index": "my_index",
+                  "_id": "1",
+                  "_nested": {
+                    "field": "comments",
+                    "offset": 0
+                  },
+                  "docvalue_fields": {
+                    "username": ["alice"],
+                    "created_at": ["2023-01-01T12:00:00Z"]
+                  }
+                },
+                {
+                  "_index": "my_index",
+                  "_id": "1",
+                  "_nested": {
+                    "field": "comments",
+                    "offset": 1
+                  },
+                  "docvalue_fields": {
+                    "username": ["bob"],
+                    "created_at": ["2023-01-02T12:00:00Z"]
+                  }
+                }
+              ]
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+## Searching with `stored_fields` 
+
+`stored_fields` is another feature in OpenSearch that allows you to explicitly store and retrieve specific fields from documents, separate from the `_source` field. By default, OpenSearch stores the entire document in the `_source` field and uses it to return document contents in search results. However, sometimes you might want to store certain fields separately for more efficient retrieval.
+
+Unlike `_source`, `stored_fields` must be explicitly defined in the mappings for fields you want to store separately. It can be useful if you frequently need to retrieve only a small subset of fields and want to avoid retrieving the entire `_source` field. See following example.
```

Comment:
```suggestion
Unlike `_source`, `stored_fields` must be explicitly defined in the mappings for fields you want to store separately. It can be useful if you frequently need to retrieve only a small subset of fields and want to avoid retrieving the entire `_source` field. The following example shows using `stored_fields`.
```